### PR TITLE
When setting options as a non-associative array, Column fails with "Call to undefined method Phinx\Db\Table\Column::set0()"

### PR DESCRIPTION
**Pull request by f3ath**
This check makes sure the developer gets an appropriate message.

### CODE REVIEW

**Comment by rquadling** [comment]
Should the `Index` and `ForeignKey` also have the same guarding?

File: src/Phinx/Db/Table/ForeignKey.php [hunk 1]
```diff
@@ -216,8 +216,8 @@ public function setOptions($options)
         // Valid Options
         $validOptions = array('delete', 'update', 'constraint');
         foreach ($options as $option => $value) {
-            if (!in_array($option, $validOptions)) {
-                throw new \RuntimeException('\'' . $option . '\' is not a valid foreign key option.');
+            if (!in_array($option, $validOptions, true)) {
+                throw new \RuntimeException(sprintf('"%s" is not a valid foreign key option.', $option));
```
Comment (f3ath) [inline src/Phinx/Db/Table/ForeignKey.php]: Unified the exception messages

File: src/Phinx/Db/Table/Index.php [hunk 2]
```diff
@@ -165,8 +165,8 @@ public function setOptions($options)
         // Valid Options
         $validOptions = array('type', 'unique', 'name', 'limit');
         foreach ($options as $option => $value) {
-            if (!in_array($option, $validOptions)) {
-                throw new \RuntimeException('\'' . $option . '\' is not a valid index option.');
+            if (!in_array($option, $validOptions, true)) {
+                throw new \RuntimeException(sprintf('"%s" is not a valid index option.', $option));
```
Comment (f3ath) [inline src/Phinx/Db/Table/Index.php]: Unified the exception messages

**Comment by f3ath** [comment]
@rquadling done, with a cleaner solution

**Comment by rquadling** [comment]
Excellent work. Thank you.

**Pull request merged**